python/pyspark/sql/functions.py

```diff
@@ -2112,7 +2113,7 @@ def wrapper(*args):
 
 
 @since(1.3)
-def udf(f=None, returnType=StringType()):
+def udf(f=None, returnType=StringType(), vectorized=False):
```

Comment:
@felixcheung does this fit your idea for a more generic decorator?  Not exclusively labeled as `pandas_udf`, just enable vectorization with a flag, e.g. `@udf(DoubleType(), vectorized=True)`

Reply:
I think `@pandas_udf(DoubleType())` is better than `@udf(DoubleType(), vectorized=True)`, which is more concise.

Reply:
as we discussed in the email, we should also accept data type of string format.

Reply:
and also `**kwargs` to bring the size information

Reply:
It seems like the consensus is for `pandas_udf` and I'm fine with that too.  I'll make that change and the others brought up here.